Client/WebFilters/ContentBlocker/TrackingProtectionPageStats.swift

```diff
@@ -54,7 +54,7 @@ class TPStatsBlocklistChecker {
     static let shared = TPStatsBlocklistChecker()
     private let adblockSerialQueue = AdBlockStats.adblockSerialQueue
 
-    func isBlocked(request: URLRequest, domain: Domain, resourceType: TPStatsResourceType? = nil, _ completion: @escaping (BlocklistName?) -> Void) {
+    func isBlocked(tab: Tab?, request: URLRequest, domain: Domain, resourceType: TPStatsResourceType? = nil, _ completion: @escaping (BlocklistName?) -> Void) {
```

Comment:
Is there a reason for this `Tab` to be optional?

Reply:
Why don't we just pass the `mainDocumentUrl` instead of the tab itself? We already have it unwrapped and guarded earlier in the code

Reply:
Ah, I was just using the existing code there which uses the `tab.url`. The `tab` is defined as `weak tab: Tab?` so it is optional and I just passed it right in.

I have to check if `mainDocumentUrl` is the same as `tab.url`.